vault/dynamic_system_view.go

```diff
@@ -86,11 +86,7 @@ func (d dynamicSystemView) CachingDisabled() bool {
 
 // Checks if this is a primary Vault instance.
 func (d dynamicSystemView) ReplicationState() consts.ReplicationState {
-	var state consts.ReplicationState
-	d.core.clusterParamsLock.RLock()
-	state = d.core.replicationState
-	d.core.clusterParamsLock.RUnlock()
-	return state
+	return d.core.replicationState
```

Comment:
Although I doubt this is being accessed not in the context of a request (and thus holding the read lock) I think it'd be worth calling that out specifically in the comment that it's required.